pkg/gzip/gzip.go

```diff
@@ -165,3 +191,10 @@ func ExtractTarGz(owner string, fp io.Reader, dest string) (err error) {
 	}
 	return nil
 }
+
+func firstExistingParent(path string) string {
+	for _, err := os.Stat(path); err != nil; _, err = os.Stat(path) {
```

Comment:
then it will not be the case that `err != nil`, thus the loop will not terminate, thus we will go one path higher up. Is it possible for stat to fail without knowing what's up... I'm not sure.

If stat could fail intermittently, I'm a bit concerned because it could falsely allow a bad tarball through. If tarball has `dir/passwd`, but statting `dir` fails for whatever reason, even though `dir` links to `/etc`, then we could overwrite `/etc/passwd`.